plugins/snowflake/dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -0,0 +1,85 @@
+
+{% materialization incremental, adapter='snowflake' -%}
+
+  {%- set unique_key = config.get('unique_key') -%}
+  {%- set sql_where = config.get('sql_where') -%}
+
+  {%- set non_destructive_mode = (flags.NON_DESTRUCTIVE == True) -%}
+  {%- set full_refresh_mode = (flags.FULL_REFRESH == True) -%}
+
+  {%- set identifier = model['alias'] -%}
+
+  {%- set old_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) -%}
+
+  {%- set target_relation = api.Relation.create(database=database, identifier=identifier, schema=schema, type='table') -%}
+
+  {%- set exists_as_table = (old_relation is not none and old_relation.is_table) -%}
+  {%- set exists_not_as_table = (old_relation is not none and not old_relation.is_table) -%}
+
+
+  {%- set force_create_or_replace = full_refresh_mode -%}
+
+
+  -- setup
+
+  {% set source_sql -%}
+     {#-- wrap sql in parens to make it a subquery --#}
+     (
+        select * from (
+            {{ sql }}
+        )
+        {% if sql_where %}
+            where ({{ sql_where }}) or ({{ sql_where }}) is null
+        {% endif %}
+    )
+  {%- endset -%}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+  -- `BEGIN` happens here:
+  {{ run_hooks(pre_hooks, inside_transaction=True) }}
+
+  -- build model
+  {% if force_create_or_replace or old_relation is none -%}
+    {%- call statement('main') -%}
+
+      {# -- create or replace logic because we're in a full refresh or table is non existant. #}
+      {% if old_relation is not none and old_relation.type == 'view' %}
```

Comment:
this pattern leads to downtime if a model is switched from a view to a table, but we do something similar for other adapters too. I think that's ok, as it's a one-time thing. Just wanted to call it out

Reply:
yeah good you point it out and I don't have an alternative since I think in that case there's no other way no to drop but I think it really makes sense and user would expect downtime intuitively so for now let's leave it as is and if some more global solution comes up then it makes sense to modify other adaptors